valohai_yaml/utils/lint.py

```diff
@@ -17,3 +25,27 @@ def lint_iterables(
         for item in iterable:
             if hasattr(item, "lint"):
                 item.lint(lint_result, context)
+
+
+def lint_expression(
+    lint_result: "LintResult",
+    context: LintContext,
+    field_name: str,
+    expression: Optional[str],
+) -> None:
+    if expression is not None:
+        universe = VerifierUniverse()
+        evl = _KeywordRewriterEvaluator(universe, max_depth=8)
+        for k in context:
+            if k in ["task", "pipeline", "step"]:
+                object_type = k
+                break
+        else:
+            return
+        try:
+            evl.evaluate_expression(expression)
+        except Exception as e:
+            lint_result.add_error(
+                f"{object_type.capitalize()} {context[object_type].name}, "
+                f"`{field_name}` is not a valid expression: {e}",
+            )
```

Comment:
the suggestion is opinionated, use parts that you might more clean, or none at all :stuck_out_tongue: 

mainly declaring things only as they are needed and reducing the nesting

```suggestion
def lint_expression(
    lint_result: "LintResult",
    context: LintContext,
    field_name: str,
    expression: Optional[str],
) -> None:
    if expression is None:
        return

    object_type = None
    for key in context:
        if key in ["task", "pipeline", "step"]:
            object_type = key
            break

    if object_type is None:
        return

    universe = VerifierUniverse()
    evaluator = _KeywordRewriterEvaluator(universe, max_depth=8)
    try:
            evaluator.evaluate_expression(expression)
    except Exception as e:
        lint_result.add_error(
            f"{object_type.capitalize()} {context[object_type].name}, "
            f"`{field_name}` is not a valid expression: {e}",
        )
```

Reply:
I touched it up a tad and removed the mystery meat for loop to get the object type (the object should pass it in the context instead)